src/app/core/fleets/fleet_configuration.nim

```diff
@@ -39,7 +39,7 @@ proc newFleetConfiguration*(jsonString: string): FleetConfiguration =
 proc delete*(self: FleetConfiguration) =
   discard
 
-proc extractConfig(self: FleetConfiguration, jsonString: string) =
+proc extractConfig(self: FleetConfiguration, jsonString: string) {.gcsafe.} =
```

Comment:
Why was gcsafe needed?

Reply:
@jrainville the thing here is that new `FleetConfiguration` instance is created from the `MailserverModel` and that model is being executing in a separate thread, now when if `newFleetConfiguration` constructors contains code which is in `extractConfig` all would be good, since compiler is able to infer the intention and knows that crated instance will manage local resources, but in case when we call `extractConfig` from the constructor compiler is not able to infer our intention even that method is private, it looks at it like some aync call and in that case `{.gcsafe.}` pragma is mandatory to tell to compiler what is our intention.